### ISSUE TITLE
What loss is suitable for one anchor, multiple positive and multiple negative？

### ISSUE THREAD

**Issue by ImmortalSdm** [KevinMusgrave/pytorch-metric-learning#693]

**Comment by KevinMusgrave** [comment]
Apologies for the late reply.

You can use the concept of `ref_emb` to separate anchors from positives and negatives. 

For example, using ContrastiveLoss:

```python
from pytorch_metric_learning.losses import ContrastiveLoss

loss_fn = ContrastiveLoss()

# anchors has shape NxD
# anchor_labels has shape N
# ref_emb has shape MxD
# ref_labels has shape M
loss = loss_fn(anchors, anchor_labels, ref_emb=ref_emb, ref_labels=ref_labels)
```
